lighthouse-core/lib/url-shim.js

```diff
@@ -96,6 +96,19 @@ URL.getURLDisplayName = function getURLDisplayName(url, options) {
   return Util.getURLDisplayName(new URL(url), options);
 };
 
+/**
+ * @param {string} url
+ * @return {string}
+ */
+URL.elideDataURI = function elideDataURI(url) {
+  try {
+    const parsed = new URL(url);
+    return parsed.protocol === 'data:' ? url.slice(0, 100) : url;
```

Comment:
that seems slightly more confusing to me since I started second guessing if parsed.href is somehow different from `url`, is there a reason you prefer it?